src/Core/AdminConsole/OrganizationFeatures/Policies/PolicyValidators/OrganizationDataOwnershipPolicyValidator.cs

```diff
@@ -17,28 +16,45 @@ public class OrganizationDataOwnershipPolicyValidator(
     IEnumerable<IPolicyRequirementFactory<IPolicyRequirement>> factories,
     IFeatureService featureService,
     ILogger<OrganizationDataOwnershipPolicyValidator> logger)
-    : OrganizationPolicyValidator(policyRepository, factories)
+    : OrganizationPolicyValidator(policyRepository, factories), IPostSavePolicySideEffect, IPolicyValidator
 {
     public override PolicyType Type => PolicyType.OrganizationDataOwnership;
 
     public override IEnumerable<PolicyType> RequiredPolicies => [];
 
     public override Task<string> ValidateAsync(PolicyUpdate policyUpdate, Policy? currentPolicy) => Task.FromResult("");
 
-    public override async Task OnSaveSideEffectsAsync(PolicyUpdate policyUpdate, Policy? currentPolicy)
+    public async Task ExecuteSideEffectsAsync(SavePolicyModel policyRequest, Policy? postUpdatedPolicy)
     {
         if (!featureService.IsEnabled(FeatureFlagKeys.CreateDefaultLocation))
         {
             return;
         }
 
-        if (currentPolicy?.Enabled != true && policyUpdate.Enabled)
+        if (policyRequest.Metadata is not OrganizationModelOwnershipPolicyModel metadata)
+        {
+            return;
+        }
+
+        if (string.IsNullOrWhiteSpace(metadata.DefaultUserCollectionName))
+        {
+            return;
+        }
+
+        var policyUpdate = policyRequest.PolicyUpdate;
+
+        if (postUpdatedPolicy?.Enabled == true || !policyUpdate.Enabled)
         {
-            await UpsertDefaultCollectionsForUsersAsync(policyUpdate);
+            return;
         }
```

Comment:
We want to run the side effect if the policy is changing state from disabled -> enabled. However, this method only has access to the policy **request** and the policy **after** it's been updated; you also need to know the previous state to know whether it's been toggled.

The old validator interface got the `currentPolicy` passed into it so it could determine this.

A quick fix for this would just be to return the `currentPolicy` object out of `SavePolicyCommand.SaveAsync`, e.g. `public async Task<(Policy PostUpdate, Policy PreUpdate)> SaveAsync(PolicyUpdate policyUpdate)`. That way it's in scope in your vnext method.